H2MLauncher.UI/ViewModels/GameStateViewModel.cs

```diff
@@ -0,0 +1,111 @@
+﻿using System.Net;
+
+using CommunityToolkit.Mvvm.ComponentModel;
+
+using H2MLauncher.Core.Services;
+
+namespace H2MLauncher.UI.ViewModels;
+
+public partial class GameStateViewModel : ObservableObject
+{
+    [ObservableProperty]
+    private bool _isGameRunning;
+
+    [ObservableProperty]
+    private bool _isConnected;
+
+    public int? ProcessId => DetectedGame?.Process.Id;
+
+    [ObservableProperty]
+    private string _displayText = "Game not detected";
+
+    [ObservableProperty]
+    [NotifyPropertyChangedFor(nameof(ConnectionText))]
+    private string? _connectedIp;
+
+
+    private GameState? _state;
+    public GameState? State
+    {
+        get
+        {
+            return _state;
+        }
+        set
+        {
+            _state = value;
+            ConnectedIp = _state?.Endpoint?.Address.ToString();
+            IsConnected = _state?.IsConnected ?? false;
+            OnPropertyChanged(nameof(ConnectionText));
+        }
+    }
+
+    public string ConnectionText
+    {
+        get
+        {
+            if (State is null)
```

Comment:
I have the feeling you can just use pattern matching here

Reply:
Oh right true, i will try that 👍